src/v/datalake/local_parquet_file_writer.cc

```diff
@@ -101,7 +101,7 @@ ss::future<writer_error> local_parquet_file_writer::flush() {
     if (!_initialized) {
         co_return writer_error::flush_error;
     }
-    if (_error != writer_error::ok) {
+    if (_error != writer_error::ok && !is_recoverable_error(_error)) {
```

Comment:
think a similar change is also needed in L133 and L136

Reply:
yep, thanks. fixed